spec/relativetimeformat.html

```diff
@@ -1,61 +1,256 @@
+        1. Let _localeData_ be *%RelativeTimeFormat%*.[[LocaleData]].
+        1. Let _r_ be ResolveLocale(*%RelativeTimeFormat%*.[[AvailableLocales]], _requestedLocales_, _opt_, *%RelativeTimeFormat%*.[[RelevantExtensionKeys]], _localeData_).
+        1. Set _relativeTimeFormat_.[[Locale]] to the value of _r_.[[Locale]].
+        1. Let _dataLocale_ be _r_.[[DataLocale]].
         1. Let _s_ be ? GetOption(_options_, *"style"*, *"string"*, «*"long"*, *"short"*, *"narrow"*», *"long"*).
-        1. Set _relativeTimeFormat_.[[style]] to _s_.
+        1. Set _relativeTimeFormat_.[[Style]] to _s_.
         1. Let _u_ be ? GetOption(_options_, *"unit"*, *"string"*, «*"best fit"*, *"second"*, *"minute"*, *"hour"*, *"day"*, *"week"*, *"month"*, *"quarter"*, *"year"*», *"best fit"*).
-        1. Set _relativeTimeFormat_.[[unit]] to _u_.
+        1. Set _relativeTimeFormat_.[[Unit]] to _u_.
         1. Let _dataLocaleData_ be Get(_localeData_, _dataLocale_).
         1. Let _fields_ be Get(_dataLocaleData_, *"fields"*).
         1. Assert: _fields_ is an object (see <emu-xref href="#sec-Intl.RelativeTimeFormat-internal-slots"></emu-xref>).
-        1. Set _relativeTimeFormat_.[[fields]] to _fields_.
-        1. Let _pluralRules_ be Get(_dataLocaleData_, *"pluralRules"*).
-        1. Assert: _pluralRules_ is an object (see <emu-xref href="#sec-Intl.RelativeTimeFormat-internal-slots"></emu-xref>).
-        1. Set _relativeTimeFormat_.[[pluralRule]] to Get(_pluralRules_, *"cardinal"*).
-        1. Let _nf_ be Construct(%NumberFormat%, «[_r_.[[locale]], {minimumIntegerDigits: 2, useGrouping: *false*}»).
-        1. Set _relativeTimeFormat_.[[numberFormat]] to _nf_.format.
-        1. Set _relativeTimeFormat_.[[boundResolve]] to *undefined*.
-        1. Set _relativeTimeFormat_.[[initializedRelativeTimeFormat]] to *true*.
+        1. Set _relativeTimeFormat_.[[Fields]] to _fields_.
+        1. Let _nf_ be Construct(%NumberFormat%, «[_r_.[[Locale]], {minimumIntegerDigits: 2, useGrouping: *false*}»).
+        1. Set _relativeTimeFormat_.[[NumberFormat]] to _nf_.format.
+        1. Let _pr_ be Construct(%PluralRules%, «[_r_.[[Locale]], {style: *"cardinal"*}»).
+        1. Set _relativeTimeFormat_.[[PluralRules]] to _pr_.select.
+        1. Set _relativeTimeFormat_.[[InitializedRelativeTimeFormat]] to *true*.
         1. Return _relativeTimeFormat_.
       </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-PartitionRelativeTimePattern" aoid="PartitionRelativeTimePattern">
+      <h1>PartitionRelativeTimePattern (relativeTimeFormat, x)</h1>
+
+      <p>
+        When the FormatRelativeTime abstract operation is called with arguments _relativeTimeFormat_ (which must be an object initialized as a RelativeTimeFormat) and _value_ (which must be a Number value), it returns a String value representing _value_ (interpreted as a time value as specified in ES2015, 20.3.1.1) according to the effective locale and the formatting options of _relativeTimeFormat_.
```

Comment:
I also prefer to have the assertions implicitly in the algo rather than wording on this paragraph.